### ISSUE TITLE
Pharo7 - seg fault loading a baseline from command line, suspect some libgit timing error

### ISSUE THREAD

**Issue by macta** [guillep/pharo-core#2582]
I have been using zerconf fairly extensively in testing Exercism/pharo - and our instructions are:

```
curl https://get.pharo.org | bash

./pharo-ui Pharo.image eval "
Metacello new 
 baseline: 'Exercism'; 
 repository: 'github://macta/pharo-exercism:master/dev/src';
 load.
ExercismManager welcome.
"
```
[crash.dmp.log](https://github.com/pharo-project/pharo/files/2873650/crash.dmp.log)

This works fairly well - but I just got a seg fault running this which might usefully point to some libgit timing/coding error? When my baseline was loading, I noticed it got stalled a few times on some of the dependencies and so it was taking longer than normal (e.g. some of the sub loads which take 1s, were taking 5-10s). Before the full baseline completed the image terminated and the crash.dmp file is attached in case it helps.

**Comment by macta** [comment]
This is on OSX HighSierra, with Pharo7 (32 bit - as I think zero conf defautls to that),

**Comment by Ducasse** [comment]
Hi tim
I'm also on highsierra but 64 VM. What I noticed is that images takes time to go back from sleep
but do not crash. 
Stef

**Comment by macta** [comment]
@Ducasse - yes I also get the longish beach balls on some images when coming out of sleep (e.g. 20s or more) - but I recall that also in P6 (and I think I might have reported it). In these cases they do seem to come back. This said, this specific example is not sleep related, its a crash from simple command line execution - in the course of 3 minutes.

Interestingly, the other crashes I have reported in #2422 actually seemed to have occurred when the laptop was sleeping (looking at the crash dumps, my laptop was closed but on power at the time the crash occured - so something was running)
